flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/sort/BaseTemporalSortOperator.java

```diff
@@ -38,6 +39,7 @@ abstract class BaseTemporalSortOperator extends AbstractStreamOperator<RowData>
 
     @Override
     public void open() throws Exception {
+        setChainingStrategy(ChainingStrategy.ALWAYS);
```

Comment:
isn't that a bit too late? that's runtime already, no?

Reply:
Absolutely it is.